crates/red_knot_python_semantic/resources/mdtest/boundness_declaredness/public.md

```diff
@@ -106,8 +111,8 @@ if flag():
 ```py
 from mod import x, y
 
-reveal_type(x)  # revealed: Literal[1] | Any
-reveal_type(y)  # revealed: Literal[2] | Unknown
+reveal_type(x)  # revealed: Unknown | Literal[1] | Any
```

Comment:
We should probably cancel one of the dynamic types here? We might have a ticket for that. I'll note it down as a follow up.

Reply:
I think we have a TODO in `Type::is_gradual_equivalent_to` pointing out that not doing this causes problems... but I don't know that we have a ticket

Reply:
I agree that we should do that, but it will also make this test less clearly testing what it is supposed to test. The reason we use `x: Any` here is that it a) doesn't disappear in union with `Literal[1]` but b) also doesn't cause an invalid-declaration diagnostic.

But we can discuss how to handle this in the follow up PR.